tests/test_config/test_snapshot.py

```
@@ -0,0 +1,190 @@
+    def test_env_refs_preserved(self):
+        raw = {"services": {"m": {"api_key": "${NVIDIA_API_KEY}"}}}
+        assert mask_secrets(raw) == raw
+
+    def test_literal_secret_key_masked(self):
+        raw = {"services": {"m": {"api_key": "nvapi-not-a-real-key"}}}
+        assert mask_secrets(raw)["services"]["m"]["api_key"] == "<redacted>"
+
+    def test_env_var_names_masked_when_literal(self):
+        raw = {"env_vars": {"HF_TOKEN": "hf_literalvalue", "JUDGE_API_KEY": "${JUDGE_API_KEY}"}}
+        out = mask_secrets(raw)
+        assert out["env_vars"]["HF_TOKEN"] == "<redacted>"  # key matches token
```

Ruff (0.15.20)

[error] Possible hardcoded password assigned to: "HF_TOKEN"

(S105)